static/js/narrow.js

```diff
@@ -749,6 +749,23 @@ exports.restore_home_state = function () {
     navigate.maybe_scroll_to_selected();
 };
 
+function show_search_query() {
+    var search_query = $("#search_query").val();
+    var query_words = search_query.split(" ");
+    var filtered_words = "";
+    _.each(query_words, function (word) {
+        var stop_word = _.find(page_params.stop_words, function (stop_word) {
+            return stop_word === word;
+        });
+        if (stop_word) {
+            filtered_words += stop_word.strike().italics() + " ";
+        } else {
+            filtered_words += word + " ";
+        }
+    });
```

Comment:
Don't construct HTML by concatenating user input with other things.  That approach will always have security issues if you put HTML inside the search query.  We should use a safe escaping mechanism to prevent that.

Reply:
Can you use the approach Steve suggested here?  https://chat.zulip.org/#narrow/stream/101-design/topic/Indicate.20search.20stop.20words/near/690833  

That's cleaner and doesn't involved adding a new semi-duplicated escape function.

Reply:
I used Steve's suggestion, this took more time than it should have because I had some Casper tests failing randomly, also, thank you for reviewing!